src/MongoDB/Driver/WriteResult.cpp

```diff
@@ -197,16 +182,13 @@ Object hippo_write_result_init(mongoc_write_result_t *write_result, bson_error_t
 	}
 
 	if (success == 0) {
-		if (
-			bson_empty0(&write_result->writeErrors) &&
-			bson_empty0(&write_result->writeConcernErrors)
-		) {
-			throw MongoDriver::Utils::throwExceptionFromBsonError(error);
-		} else {
-			auto bw_exception = MongoDriver::Utils::throwBulkWriteException(message);
+		if (error->domain == MONGOC_ERROR_COMMAND || error->domain == MONGOC_ERROR_WRITE_CONCERN) {
+			auto bw_exception = MongoDriver::Utils::throwBulkWriteException(error->message);
```

Comment:
No, the else case below will take care of it:

    throw MongoDriver::Utils::throwExceptionFromBsonError(error);